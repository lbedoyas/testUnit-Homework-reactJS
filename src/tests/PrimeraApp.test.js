import { render } from "@testing-library/react";
import React from "react";
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom'
import { shallow } from "enzyme";

describe('Pruebas en primera app', () => {
    // test('debe de mostrar el mensaje "Hola soy goku" ', () => {
    //     const saludo = "Hola, Soy Goku!";
    //     const {getByText} = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe de mostrar primera app correctamente ', () => {
        const saludo = "Hola, Soy Goku";
        const wrapper = shallow(<PrimeraApp saludo = {saludo} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el subtitulo enviado por props ', () => {
        const saludo = "Hola, Soy Goku";
        const subtitulo = `Soy un subtitulo`;
        const wrapper = shallow(
            <PrimeraApp
                saludo = {saludo}
                subtitulo = {subtitulo}  />
        );
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo.trim()).toBe(subtitulo.trim());
    });
    
});