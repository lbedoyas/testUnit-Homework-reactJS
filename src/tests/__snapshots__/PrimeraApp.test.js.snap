// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera app Debe de mostrar primera app correctamente  1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
     
    Soy un subtitulo
     
  </p>
</Fragment>
`;
